infrastructure_targets.go

```diff
@@ -0,0 +1,236 @@
+package cloudflare
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/goccy/go-json"
+)
+
+var ErrMissingTargetId = errors.New("required target id missing")
+
+// Target represents an Infrastructure Target.
+type Target struct {
+	Hostname   string    `json:"hostname"`
+	ID         string    `json:"id"`
+	IP         IPInfo    `json:"ip"`
+	CreatedAt  time.Time `json:"created_at"`
+	ModifiedAt time.Time `json:"modified_at"`
+}
+
+type IPDetails struct {
+	IpAddr           string `json:"ip_addr"`
+	VirtualNetworkId string `json:"virtual_network_id"`
+}
+
+type IPInfo struct {
+	IPV4 *IPDetails `json:"ipv4,omitempty"`
+	IPV6 *IPDetails `json:"ipv6,omitempty"`
+}
+
+type InfrastructureTargetParams struct {
+	Hostname string `json:"hostname"`
+	IP       IPInfo `json:"ip"`
+}
+
+type CreateInfrastructureTargetParams struct {
+	InfrastructureTargetParams
+}
+
+type UpdateInfrastructureTargetParams struct {
+	ID           string                     `json:"hostname"`
+	ModifyParams InfrastructureTargetParams `json:"modify_params"`
+}
+
+// TargetDetailResponse is the API response, containing a single target.
+type TargetDetailResponse struct {
+	Result Target `json:"result"`
+	Response
+}
+
+type TargetListDetailResponse struct {
+	Result []Target `json:"result"`
+	Response
+	ResultInfo `json:"result_info"`
+}
+
+type TargetListParams struct {
+	CreatedAfter     string `url:"created_after,omitempty"`
+	Hostname         string `url:"hostname,omitempty"`
+	HostnameContains string `url:"hostname_contains,omitempty"`
+	IPV4             string `url:"ip_v4,omitempty"`
+	IPV6             string `url:"ip_v6,omitempty"`
+	ModifedAfter     string `url:"modified_after,omitempty"`
+	VirtualNetworkId string `url:"virtual_network_id,omitempty"`
+
+	ResultInfo
+}
+
+// ListInfrastructureTargets returns all applications within an account or zone.
+//
+// Account API reference: https://developers.cloudflare.com/api/operations/infra-targets-list
+func (api *API) ListInfrastructureTargets(ctx context.Context, rc *ResourceContainer, params TargetListParams) ([]Target, *ResultInfo, error) {
+	if rc.Identifier == "" {
+		return []Target{}, &ResultInfo{}, ErrMissingAccountID
+	}
+
+	baseURL := fmt.Sprintf("/%s/%s/infrastructure/targets", rc.Level, rc.Identifier)
+
+	autoPaginate := true
+	if params.PerPage >= 1 || params.Page >= 1 {
+		autoPaginate = false
+	}
+
+	if params.PerPage < 1 {
+		params.PerPage = 25
+	}
+
+	if params.Page < 1 {
+		params.Page = 1
+	}
+
+	var applications []Target
```

Comment:
nit: targets

Reply:
generally you want a collection of singular names. i.e. `things := []Thing`